lib/topology/platforms/shell.py

```diff
@@ -506,32 +515,37 @@ def get_response(self, connection=None, silent=False):
         # Get connection
         spawn = self._get_connection(connection)
 
-        # Convert binary representation to unicode using encoding
-        text = spawn.before.decode(self._encoding)
+        response = ''
+        try:
```

Comment:
Please explain the purpose of this change. I have to reject it because it breaks the return type of the method (that should be ``str``) and the ``try`` block is too large, it covers many lines and since the ``except`` will be triggered by any kind of exception any failure in the ``try`` block wil return ``response.before``.